csharp/src/Chunk.cs

```diff
@@ -0,0 +1,93 @@
+﻿// This file is part of the ArmoniK project
+//
+// Copyright (C) ANEO, 2022-2022.All rights reserved.
+//   W.Kirschenmann   <wkirschenmann@aneo.fr>
+//   J.Gurhem         <jgurhem@aneo.fr>
+//   D.Dubuc          <ddubuc@aneo.fr>
+//   L.Ziane Khodja   <lzianekhodja@aneo.fr>
+//   F.Lemaitre       <flemaitre@aneo.fr>
+//   S.Djebbar        <sdjebbar@aneo.fr>
+//   J.Fonseca        <jfonseca@aneo.fr>
+//   D.Brasseur       <dbrasseur@aneo.fr>
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Diagnostics;
+
+namespace ArmoniK.Utils;
+
+internal static class Chunk
+{
+  // Implementation of the AsChunked function
+  // Original source code : https://github.com/dotnet/runtime/blob/main/src/libraries/System.Linq/src/System/Linq/Chunk.cs
```

Comment:
This code is under MIT license, not Apache. 
Not sure how to deal with that...

Reply:
I'd suggest to rewrite it in a style that better fit our approaches. 
the link can be kept to explain that the algorithm has been inspired by the one from .Net.

Reply:
I'm not sure that in our case we'd loose a lot of performance by using a `List<>` and relying on the growth algorithm from `List<>`.

Reply:
I'm not sure we lose too much memory by just allocating directly an array of the chunk size. If I have to reimplement it, I think I will just do that.